cache/local_session.cpp

```diff
@@ -313,28 +313,47 @@ int local_session::remove(const struct dnet_id &id, dnet_access_context *context
 }
 
 int local_session::remove_new(const struct dnet_id &id,
-                              const ioremap::elliptics::dnet_remove_request &request,
+                              const ioremap::elliptics::n2::remove_request &request,
                               dnet_access_context *context) {
-	const auto packet = ioremap::elliptics::serialize(request);
+	using namespace ioremap::elliptics;
 
 	struct dnet_cmd cmd;
 	memset(&cmd, 0, sizeof(struct dnet_cmd));
 	cmd.id = id;
-	cmd.size = packet.size();
 	cmd.flags = DNET_FLAGS_NOLOCK;
 	cmd.cmd = DNET_CMD_DEL_NEW;
 	cmd.backend_id = m_backend.backend_id();
 
 	struct dnet_cmd_stats cmd_stats;
 	memset(&cmd_stats, 0, sizeof(struct dnet_cmd_stats));
 
+	int status = 0;
+	n2_repliers repliers{
+		[](const std::shared_ptr<n2_body> &/*body*/) {
+			return 0;
+		}, // on_reply
+		[&status](int err) {
+			status = err;
+			return 0;
+		} // on_reply_error
+	};
+
+	n2_request_info request_info{
+		{cmd, n2::default_deadline()},
+		std::move(repliers)
+	};
+	request_info.request.body = std::make_shared<n2::remove_request>(request);
```

Comment:
`request` was originally a `shared_ptr`. So it is better to pass it to this place as a `shared_ptr`.